src/logic/calculate.js

```diff
@@ -0,0 +1,25 @@
+import operate from './operate';
+
+const calculate = (data, name) => {
+  let result;
+
+  switch (name) {
+    case '+/-':
+      result = data[0] * (-1);
+      break;
+    case '.':
+      result = '.';
+      break;
+    case 'AC':
+      result = '';
+      break;
+    case '=':
+      result = operate(data[0], data[1], data[2]);
+      break;
```

Comment:
the way you using your `data` (data[0], data[1], data[2]), seems to be an array,
This should be an **OBJECT** with three properties {[see requiredments](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone4/totalNextOperate.png)}